### ISSUE TITLE
dateofdelivery doesn't work

### ISSUE THREAD

**Issue by rdy4ever** [PrestaShop/PrestaShop#11427]
**Describe the bug**
The native module dateofdelivery doesn't work. I've installed it on a fresh 1.7.5.0-beta (as well as my shop on 1.7.4.3), configured it but it just doesn't show the estimated date of delivery in the order page, in $hookDisplayBeforeCarrier, as it should.

**To Reproduce**
1. Install dateofdelivery module and configure it for one carrier
2. Go to shop and add a product to cart. Go to checkout.
3. Complete all checkout steps up to delivery step
4. In delivery step, there's nothing displayed in $hookDisplayBeforeCarrier

**Screenshots**
If applicable, add screenshots or screenrecords to help explain your problem.
![image](https://user-images.githubusercontent.com/12268701/48662350-a685b600-ea89-11e8-9af5-1d497501a46d.png)
![image](https://user-images.githubusercontent.com/12268701/48662366-d339cd80-ea89-11e8-9af3-98577d99e2c0.png)

**Additionnal information**
PrestaShop version: 1.7.4.3, 1.7.5.0-beta
PHP version: N/A

**Comment by khouloudbelguith** [comment]
Hi @rdy4ever,

Thanks for your report.
Yes, we are aware of this issue.
I checked with PS 1.7.5.0beta1 & PS1.7.4.4 & the `dateofdelivery` module v2.0.1, it doesn't work.
In FO, the module does not work, it does not display delivery time according to the module Date of delivery but according to what has been defined in the Transporters tab.
@BilalDr what do you think, it is a native module?
Thanks!

**Comment by BilalDr** [comment]
Hi @rdy4ever, 
Indeed, the module [dateofdelivery ](https://github.com/PrestaShop/dateofdelivery) doesn't work on Prestashop 1.7.5 beta and 1.7.4.3.
It's a native module and it needs to be reworked to come back on Prestashop software.

**Comment by radar6** [comment]
Still awaiting this module :)

**Comment by matks** [comment]
@marionf I tried to install this module, but in my Module Catalog I cannot find it. I type "date", "delivery", I cant find it. Can you have a look ? Maybe it was disabled for recent versions

**Comment by marionf** [comment]
@matks Yes, it's only avaible from 1.5.6.1 to 1.7.1.0

![capture d'écran_2085](https://user-images.githubusercontent.com/13449658/67085609-677f4980-f19f-11e9-97ec-67f1b5c24558.png)

**Comment by matks** [comment]
@marionf Then if it is not available for PS > 1.7.1, is this issue relevant ? 🤔 Or is this issue in fact "fix the dateofdelivery module to make it work with latest PS versions" ?

**Comment by marionf** [comment]
I think the idea is to make it work with PrestaShop 1.7
If I remember well, the module wasn't displayed in FO because it has never been integrated with Classic theme
So, maybe it needs some mockups

**Comment by TristanLDD** [comment]
From a UX point of vue, displaying the date of delivery in the checkout process is a strong 
reinsurance element, so in my opinion this module has a real value in terms of conversion. 

I can make new mockups for classic theme but first we should take the decision whether or not we want to fix the module for 1.7.4.3+  
@Coline is it a go or no go ? :)

**Comment by colinegin** [comment]
Yes this is a great idea, i'm all for it ! Do you think there's a lot of rework in terms of design ? 
In terms of features, I would recommend to stay iso as today, I can write down some specs based on the current documentation.
About compatibility, depending on what technical changes are done in the module, it might be 1.7.6+, but we'll discuss this with a dev.

Thanks !

**Comment by TristanLDD** [comment]
@colinegin Ok !  
Not much work needed on design, the module adds a little bit of text in the delivery part of the checkout. Looks like this in 1.6 : 
![capture d'écran_2115](https://user-images.githubusercontent.com/27484094/67846603-ca5ad400-fb01-11e9-8d95-1da32e1675ef.png)

I'll update the mockup for classic theme here soon.

**Comment by colinegin** [comment]
You are my hero !

**Comment by TristanLDD** [comment]
Here's the mockup for Classic thème: 

![Delivery Date](https://user-images.githubusercontent.com/27484094/67957535-1129f600-fbf6-11e9-8d1b-e3a866aede35.jpg)

**Comment by colinegin** [comment]
Better late than never ! 
Here are the specifications for the date of delivery module to be able to make it compatible with PS 1.7 : https://github.com/PrestaShop/prestashop-specs/blob/master/modules/dateofdelivery.md

Thanks @rdy4ever for raising this issue !

**Comment by Markosaar** [comment]
Another development wish is that delivery times should be identifiable by country or at least regionally, as they usually vary. But anyway, a really necessary module ... I'm really looking forward to it.
